howtos/redirect-rules.md

```diff
@@ -0,0 +1,160 @@
+
+Take into consideration whether you're targeting a specific leaf page with no children, or a parent page with children.
+
+#### Targeting a leaf page
+
+If the old path has no children, target the page directly, allowing only for an optional trailing slash.
+
+Example:
+
+```apache
+RewriteRule ^old/path/?$ /new/path/ [R=301,L]
+```
+
+- This rule matches `/old/path` and `/old/path/`, but not `/old/path/with/children`. The `/?` at the end of the old path specifies an optional trailing slash.
+- Avoid using `(.*)` at the end of the old path for leaf pages. There are no routes that could possibly match, and the additional complexity can be confusing.
+
+#### Targeting a parent page
+
+If the old path has children, target the parent page and any children, allowing for an optional trailing slash.
+
+Example:
+
+```apache
+RewriteRule ^old/path/?(.*)$ /new/path/$1 [R=301,L]
+```
+
+- This rule matches `/old/path`, `/old/path/`, and `/old/path/with/children/`. The `(.*)` at the end of the old path matches any characters after the parent path, and injects them into the redirect path.
+
+### Redirect to the most accurate page section
+
+If the best location for moved content is a specific section of a page, redirect to that section by including the anchor in the new path.
+
+Note that you'll need to use [the `NE` flag](https://httpd.apache.org/docs/current/rewrite/flags.html#flag_ne) to prevent the redirect from encoding the `#` character.
+
+Example:
+
+```apache
+RewriteRule ^old/path/?$ /new/path/#with-section [R=301,L,NE]
+```
+
+### Avoid redirect chains
+
+Redirect rules should redirect to the final destination, rather than another URL that redirects.
+
+Redirect chains can appear in our rules in two ways:
+
+1. Old redirect rules redirecting to new redirect rules
+
+   When adding a new rule, confirm that no previous rules map to the old path of your new rule. If they do, rewrite them so that they map instead to the new path of your new rule. For example, the second rule in this example should be rewritten to redirect to `/new/path/` instead of `/middle/path/`:
+
+   ````apache
+   RewriteRule ^middle/path/?$ /new/path/ [R=301,L]
+   RewriteRule ^old/path/?$ /middle/path/ [R=301,L]
+   ```apache
+
+   ````
+
+2. Relying on built-in Apache behavior to redirect new paths that lack a trailing slash
+
+   When a URL does not include a trailing slash, the Apache webserver will automatically redirect to the same URL with a trailing slash. This is handy for users, but when our redirects rely on it, it results in a double hop for the user. We should explicitly redirect to the correct URL, including the trailing slash.
+
+   Example of a redirect that will double-hop from `/new/path` to `/new/path/`:
+
+   ```apache
+   RewriteRule ^old/path/?$ /new/path [R=301,L]
+   ```
+
+   Example of a redirect that will not double-hop:
+
+   ```apache
+   RewriteRule ^old/path/?$ /new/path/ [R=301,L]
+   ```
+
+## Testing redirect rules
```

Comment:
This whole section was just ported from the documentation-guidelines file, with no changes.